.github/workflows/crucible-go-build.yml

```diff
@@ -37,117 +24,35 @@ jobs:
+      # (e.g. cabal complains it can't find a valid version of the "happy"
+      # tool).
       #
-      # 1. It rarely fails, going last can save some time if earlier steps fail
-      # 2. It must come after we save the Cabal cache as 'Package's Cabal/GHC
-      #    compatibility' runs `cabal clean`
-      - name: Install Nix
-        if: runner.os == 'Linux'
-        uses: cachix/install-nix-action@v16
-        with:
-          nix_path: nixpkgs=channel:21.11
-          install_url: https://releases.nixos.org/nix/nix-2.10.3/install
-          # This token usage is to avoid GH rate-limiting; see commit
-          # 1696558326a8c57e1a9f0848aaaa1b8440294954. Apparently it'll
-          # work for anyone.
-          extra_nix_config: |
-            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
-      - name: Setup Environment Vars
-        if: runner.os == 'Linux'
-        run: |
-          GHC=haskell.compiler.ghc$(echo ${{ matrix.ghc }} | sed -e s,\\.,,g)
-          case ${{ matrix.ghc }} in
-            9.4.8) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-24.05 ;;
-            9.6.5) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-24.05 ;;
-            9.8.2) GHC_NIXPKGS=github:nixos/nixpkgs/nixos-24.05 ;;
-            *)     GHC_NIXPKGS=github:nixos/nixpkgs/21.11 ;;
-          esac
-          echo NS="nix shell ${GHC_NIXPKGS}#cabal-install ${GHC_NIXPKGS}#${GHC} nixpkgs#gmp nixpkgs#zlib nixpkgs#zlib.dev" >> $GITHUB_ENV
-      - name: Package's Cabal/GHC compatibility
-        shell: bash
-        if: runner.os == 'Linux'
-        # Using setup will use the cabal library installed with GHC
-        # instead of the cabal library of the Cabal-install tool to
-        # verify the cabal file is compatible with the associated
-        # GHC cabal library version.  Cannot run configure or build,
-        # because dependencies aren't present, but a clean is
-        # sufficient to cause parsing/validation of the cabal file.
-        run: |
-          defsetup()  { echo import Distribution.Simple; echo main = defaultMain; }
-          setup_src() { if [ ! -f Setup.hs ] ; then defsetup > DefSetup.hs; fi; ls *Setup.hs; }
-          setup_bin() { echo setup.${{ matrix.ghc }}; }
-          with_ghc()  { $NS -c ${@}; }
-          (cd crucible-go; with_ghc ghc -o $(setup_bin) $(setup_src) && ./$(setup_bin) clean)
+      # This also periodically happens on MacOS builds due to a tar bug
+      # (symptom: "No suitable image found ... unknown file type, first
+      # eight bytes: 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00")
+      cache-key-prefix: 1-go
+      check: false
+      ghc: ${{ matrix.ghc }}
+      os: ${{ matrix.os }}
+      pre-hook: |
+        cp "cabal.GHC-${{ matrix.ghc }}.config" cabal.project.freeze
+
+        env \
+          SOLVER_PKG_VERSION="snapshot-20250326" \
+          BUILD_TARGET_OS="${{ matrix.os }}" \
+          BUILD_TARGET_ARCH="${{ matrix.os == macos-14 && 'ARM64' || 'X64' }}" \
```

Comment:
Note for reviewers: Unfortunately, the `runner` context is not available here. I think ideally, `what4-solvers` could host a script that could detect the current OS and architecture, and download and install the correct snapshot into a configurable directory. But IMO this local workaround is OK for right now.

Reply:
`$RUNNER_ARCH` is available, so using that.